components/CheckDocumentation/connections/BlockingQuery.tsx

```diff
@@ -0,0 +1,129 @@
+        of I/O or CPU resources, impacting overall performance.
+      </p>
+      <h4>Common Causes</h4>
+      <ul>
+        {showAutovacuum && (
+          <li>
+            <h5>Autovacuum (to prevent wraparound)</h5>
+            <p>
+              In general, autovacuum workers don't block other queries. Even if
+              the other query tries to acquire a lock that conflicts with the
+              running autovacuum, lock acquisition will interrupt the
+              autovacuum. However, if the autovacuum is running to prevent
+              transaction ID wraparound, the vacuum won't be interrupted and
+              will block other queries trying to acquire conflicting locks.
+            </p>
+          </li>
+        )}
+        <li>
+          <h5>Locks</h5>
+          <p>
+            The queries may be holding some strong locks. Check the{" "}
+            <Link to={backendsUrl}>Connection Traces</Link> page for the queries
+            these queries are blocking. You can check{" "}
+            <strong>Wait Events</strong> page for the queries waiting for these
+            blocking queries, and look for any wait events of type{" "}
+            <strong>Lock</strong>. If you have{" "}
+            <SettingLink setting="log_lock_waits" /> turned on, you can also see
+            details about locking in the <strong>Logs</strong> page for these
+            queries.
+          </p>
+          <p>
+            <code>ACCESS EXCLUSIVE</code> lock xis the strongest lock that can
+            even block a <code>SELECT</code> query. This lock is acquired by the
+            commands like <code>VACUUM FULL</code>. One of the table-rewriting
+            variants of <code>ALTER TABLE</code> also acquire a lock at this
+            level.
+          </p>
+        </li>
+        <li>
+          <h5>Long-running migration</h5>
+          <p>
+            If some of these queries are DDL that is part of a database
+            migration, they may need to rewrite a table (e.g., changing the type
+            of a column) or scan a whole table (e.g., <code>CREATE INDEX</code>{" "}
+            without <code>CONCURRENTLY</code>), which can take much longer in
+            production than in a staging or development environment. If your
+            migration is transactional, consider canceling it and refactor it to
+            avoid table rewrites.
+          </p>
+        </li>
+      </ul>
+      <h4>Solution</h4>
+      <p>
+        You can cancel these queries with{" "}
+        <SQL inline sql={`SELECT pg_cancel_backend('<query_pid>');`} />. Note
+        that this only treats the symptom: if the query runs again without any
+        changes, it is likely to cause the blocking.
+      </p>
+      {showAutovacuum && (
+        <p>
+          If you're canceling the autovacuum, please note that the autovacuum to
+          prevent wraparound is really important to run and it will most likely
+          come back.
```

Comment:
Good idea to note this separately, but maybe something like
```
          If you're canceling an autovacuum to prevent wraparound, note that it does critical work and
          it will restart and will need to complete eventually, acquiring the same locks. Consider tuning
          your vacuum settings to be more aggressive to avoid the need for explicit vacuums to prevent
          wraparound
```
?

Reply:
Thanks! Yeah I wanted to give some "advice" here, tuning the vacuum settings sounds good here to me 👍